src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java

```diff
@@ -65,6 +68,15 @@ public void triggerMatchingJobs(String user, String url, String scm) {
         }
     }
 
+    private boolean hasBeenTriggered(List<SCM> scmTriggered, SCM scmTrigger) {
+        for (SCM scm : scmTriggered) {
+            if (scm.equals(scmTrigger)) {
```

Comment:
`GitSCM` does not seem to override `equals` either. But perhaps the issue here is that an identical `SCM` instance has been passed several times.

I think this is the wrong fix altogether and should be reverted. Probably this is some special case of JENKINS-29326, and needs to be fixed in `WorkflowJob.getSCMs`.